procrastinate/periodic.py

```diff
@@ -35,10 +36,26 @@ class PeriodicTask(Generic[P, R, Args]):
     cron: str
     periodic_id: str
     configure_kwargs: tasks.ConfigureTaskOptions
+    tzinfo: str | None | ZoneInfo = None
 
     @cached_property
     def croniter(self) -> croniter.croniter:
-        return croniter.croniter(self.cron)
+        croniter_instance = croniter.croniter(self.cron)
+        # croniter sets the timezone info object in
+        croniter_instance.tzinfo = self.get_tzinfo()
+        return croniter_instance
+
+    def get_tzinfo(self):
+        tzinfo = self.tzinfo
+        if isinstance(tzinfo, str):
+            try:
+                return ZoneInfo(tzinfo)
+            except (ZoneInfoNotFoundError, ValueError):
+                logger.error(f"{tzinfo} is not a valid timezone.")
+                return None
```

Comment:
A failure could be impactful, I think we need to raise.

Reply:
> The only issue I can see is that it might confuse users when their chosen timezone isn’t applied. Besides that, does it have any other impact?

If I have scheduled batch processing at 12AM while there's no activity, and it happens at 2PM during peak hours, it could easily cause disruption.